libs/utils/src/lsn.rs

```diff
@@ -343,4 +414,64 @@ mod tests {
         assert_eq!(lsn.fetch_max(Lsn(6000)), Lsn(5678));
         assert_eq!(lsn.fetch_max(Lsn(5000)), Lsn(6000));
     }
+
+    #[test]
+    fn test_lsn_serde() {
+        let lsn = Lsn(1234);
+
+        // Testing human_readable ser/de
+        let serializer = Serializer::builder().is_human_readable(false).build();
+        assert_ok_eq!(lsn.serialize(&serializer), Tokens(vec![Token::U64(1234)]));
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(false)
+            .tokens(Tokens(vec![Token::U64(1234)]))
+            .build();
+        assert_ok_eq!(Lsn::deserialize(&mut deserializer), lsn);
+
+        // Testing NON human_readable ser/de
+        let serializer = Serializer::builder().is_human_readable(true).build();
+        assert_ok_eq!(
+            lsn.serialize(&serializer),
+            Tokens(vec![Token::Str(String::from("0/4D2"))])
+        );
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(true)
+            .tokens(Tokens(vec![Token::Str(String::from("0/4D2"))]))
+            .build();
+        assert_ok_eq!(Lsn::deserialize(&mut deserializer), lsn);
+
+        // Testing mismatching ser/de
+        let serializer = Serializer::builder().is_human_readable(false).build();
+        let f_serialized = lsn.serialize(&serializer).unwrap();
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(true)
+            .tokens(f_serialized)
+            .build();
+        assert!(Lsn::deserialize(&mut deserializer).is_err());
+
+        let serializer = Serializer::builder().is_human_readable(true).build();
+        let t_serialized = lsn.serialize(&serializer).unwrap();
+
+        let mut deserializer = Deserializer::builder()
+            .is_human_readable(false)
+            .tokens(t_serialized)
+            .build();
+        assert!(Lsn::deserialize(&mut deserializer).is_err());
```

Comment:
```suggestion
        Lsn::deserialize(&mut deserializer).unwrap_err();
```

Instead of `assert!(res.is_ok())` or `assert!(res.is_err())` always use `unwrap` and `unwrap_err`. If it's wrong, you get more context (the Debug formatting of Ok/Err) than for this boolean assertion failure.

Reply:
Done https://github.com/neondatabase/neon/pull/5335/commits/c6fae9c23c806c39e4cdd355441538506d399f7b